patches/net/minecraft/server/packs/repository/PackRepository.java.patch

```diff
@@ -9,6 +9,15 @@
      }
  
      public void reload() {
+@@ -33,7 +_,7 @@
+     }
+ 
+     private Map<String, Pack> discoverAvailable() {
+-        Map<String, Pack> map = Maps.newTreeMap();
++        Map<String, Pack> map = Maps.newLinkedHashMap(); // Neo: use a linked hash map to preserve ordering
```

Comment:
IMO we still need to sort the packs produced by each `RepositorySource` before adding them to the map. The previous code would be able to handle sources that produce packs in a non-deterministic order; this implementation cannot as it only sorts by order of sources, not the packs they produce.

However, someone more familiar with the pack handling code can correct me if the order doesn't matter.

Reply:
The entire point is to preserve ordering - blindly sorting won't really help. I can check vanilla's sources, but the output of modded sources should not be sorted IMO.

Reply:
OK you were right. Now the sorting only happens within each source. That should be acceptable (and still much better than the current behavior).